src/java/main/org/apache/zookeeper/server/ZooKeeperServer.java

```diff
@@ -303,15 +305,38 @@ public void loadData() throws IOException, InterruptedException {
 
     public void takeSnapshot(){
```

Comment:
@hanm sure, please check the [zkpipe](https://github.com/flier/zkpipe) project, we are using it to tracing and audit zookeeper operations